extensions/all-extensions/pom.xml

```diff
@@ -23,6 +23,25 @@
             <artifactId>mule-module-file</artifactId>
             <version>${project.version}</version>
             <type>zip</type>
+            <exclusions>
```

Comment:
The exclusion is just a matter of having fixed the maven-assembly-plugin to exclude every transitive dependency on these extensions.
It is still not simple to answer that question, but the goal is to only expose API from Mule and a few common third party libraries (actually their packages) in order to have a lightweight distribution.

As a golden rule we have to assume that these dependencies would not be at runtime because the runtime engine will not even have them (in a later stage whenever we have cleaned things in distribution).

What I have being doing in order to validate if the extension pom has the correct dependencies is to use maven-dependency-plugin:analyze and with the output check when I have a dependency not declared in my pom.

For instance: file extension without the change proposed here in this PR
{code}
[INFO] --- maven-dependency-plugin:2.10:analyze (default-cli) @ mule-module-file ---
[WARNING] Used undeclared dependencies found:
[WARNING]    javax.inject:javax.inject:jar:1:provided
[WARNING]    commons-io:commons-io:jar:2.4:provided
[WARNING]    org.hamcrest:hamcrest-core:jar:1.3:test
[WARNING]    org.mule.tests:mule-tests-unit:jar:4.0-SNAPSHOT:test
[WARNING]    org.mule:mule-api:jar:1.0.0-SNAPSHOT:provided
[WARNING]    org.hamcrest:hamcrest-library:jar:1.3:test
[WARNING]    commons-lang:commons-lang:jar:2.4:provided
[WARNING]    com.google.guava:guava:jar:18.0:provided
[WARNING]    junit:junit:jar:4.11:test
[WARNING]    org.mockito:mockito-core:jar:1.9.5:test
[WARNING]    org.mule.modules:mule-module-extensions-support:jar:4.0-SNAPSHOT:provided
[WARNING]    org.slf4j:slf4j-api:jar:1.7.7:provided
[WARNING] Unused declared dependencies found:
[WARNING]    com.googlecode.multithreadedtc:multithreadedtc:jar:1.01:test
[WARNING]    org.mule:mule-core-tests:test-jar:tests:4.0-SNAPSHOT:test
[WARNING]    org.mule.modules:mule-module-extensions-spring-support:jar:4.0-SNAPSHOT:provided
{code}

You have to know that these libraries are exposed from the container:
{code}
org.slf4j:slf4j-api:jar:1.7.7:provided
org.mule:mule-api:jar:1.0.0-SNAPSHOT:provided
org.mule.modules:mule-module-extensions-support:jar:4.0-SNAPSHOT:provided
{code}
But the rest of them (not the one for testing or even the extension support) has to be declared in order to get it packaged in this extension's lib folder.

I would improve the mule-extensions parent pom to fail the build if this analysis has any dependency in a plugin that is used and not declared or the other way around, excluding only the core extension support and testing stuff.
I think that it would help to avoid having issues like the one reported here?